src/ui/input_dialog.rs

```diff
@@ -0,0 +1,147 @@
+use std::rc::Rc;
+
+use crossterm::event::{KeyCode, KeyEvent};
+use log::debug;
+use ratatui::{
+    layout::{Constraint, Flex, Layout, Margin, Rect},
+    style::{Color, Style},
+    widgets::{Block, BorderType, Borders, Clear},
+    Frame,
+};
+
+use crate::{actions::MonActions, model::model::Model, traits::IWindow, ui::action::UiActions};
+
+use super::{
+    action::Action,
+    widgets::{button::ButtonElement, input_field::InputFieldElement},
+    window::Window,
+};
+
+struct InputDialogState {
+    caption: String,
+    content: String,
+    hint: String,
+}
+
+fn on_init(w: &mut Window<InputDialogState>) {
```

Comment:
Do you think that it makes sense to make input dialog as a trait? Say you want to input ip addr which has some checks to be valid, mac addr, which also has some other checks, etc. etc. It's more about near future implementations rather than current PR

Reply:
@uncleDecart if that would be so simple... PRs are always welcome ;)